Raven.Database/Bundles/Replication/Tasks/ReplicationTask.cs

```diff
@@ -87,10 +88,34 @@ public class ReplicationTask : IStartupTask, IDisposable
 
         private CancellationTokenSource _cts;
 
+        private Timer _propagationTimeoutTimer;
+        private ThreadLocal<bool> isThreadProcessingReplication = new ThreadLocal<bool>();
+
+        /// <summary>
+        /// Indicates that this thread is doing replication.
+        /// </summary>
+        internal ThreadLocal<bool> IsThreadProcessingReplication { get { return isThreadProcessingReplication; } set { isThreadProcessingReplication = value; } }
+
+        private bool CheckIfShouldWakeUpReplicationAndResetTimerIfNeeded()
+        {          
+            //If ThreadDoingReplication.Value is false it means that the source of the document isn't from replication controller,
+            //thus we should wake up replication right away.
+            if (!IsThreadProcessingReplication.Value)
+            {
+                _propagationTimeoutTimer.Stop();
+                return true;
+            }
+            _propagationTimeoutTimer.Start();
```

Comment:
I have tested this because i was not sure myself luckily it is what i needed.
If the timer started already it is a noop and if it didn't it will start.